weaver/sdks/fabric/interoperation-node-sdk/src/SatpAssetManager.ts

```diff
@@ -0,0 +1,1111 @@
+        tx.setEndorsingOrganizations(...endorsingOrgs)
+    }
+    const [result, submitError] = await helpers.handlePromise(
+        tx.submit(...ccArgs)
+    );
+    if (submitError) {
+        throw new Error(`UnlockAssetUsingContractId submitTransaction Error: ${submitError}`);
+    }
+    return result;
+};
+
+/**
+ * Rollback step of a Hashed Time Lock Contract
+ * - Reclaim a set of fungible assets
+ **/
+const reclaimFungibleAssetInHTLC = async (
+    contract: Contract,
+    contractId: string,
+    endorsingOrgs: Array<string> = []
+): Promise<any> => {
+
+    if (!contract)
+    {
+        logger.error("Contract handle not supplied");
+        return false;
+    }
+    if (!contractId)
+    {
+        logger.error("contract ID not supplied");
+        return false;
+    }
+
+    // Normal invoke function
+    const tx = contract.createTransaction("UnlockFungibleAsset")
+    const ccArgs = [contractId]
+    if (endorsingOrgs && endorsingOrgs.length > 0) {
+        tx.setEndorsingOrganizations(...endorsingOrgs)
+    }
+    const [result, submitError] = await helpers.handlePromise(
+        tx.submit(...ccArgs)
+    );
+    if (submitError) {
+        throw new Error(`UnlockFungibleAsset submitTransaction Error: ${submitError}`);
+    }
+    return result;
+};
+
+/**
+ * Query the state of a Hashed Time Lock Contract
+ * - Determine if a unique asset instance is locked by a given party for another given party
+ **/
+const isAssetLockedInHTLC = async (
+    contract: Contract,
+    assetType: string,
+    assetID: string,
+    recipientECert: string,
+    lockerECert: string,
+): Promise<any> => {
+
+    if (!contract)
+    {
+        logger.error("Contract handle not supplied");
+        return false;
+    }
+    if (!assetType)
+    {
+        logger.error("Asset type not supplied");
+        return false;
+    }
+    if (!assetID)
+    {
+        logger.error("Asset ID not supplied");
+        return false;
+    }
+    if (!recipientECert)
+    {
+        logger.error("Recipient ECert not supplied");
+        return false;
+    }
+    if (!lockerECert)
+    {
+        logger.error("Locker ECert not supplied");
+        return false;
+    }
+
+    const assetExchangeAgreementStr = createAssetExchangeAgreementSerialized(assetType, assetID, recipientECert, lockerECert);
+
+    // Normal invoke function
+    const [result, evaluateError] = await helpers.handlePromise(
+        contract.evaluateTransaction("IsAssetLocked", assetExchangeAgreementStr),
+    );
+    if (evaluateError) {
+        throw new Error(`IsAssetLocked evaluateTransaction Error: ${evaluateError}`);
+    }
+    return result.toString() === "true";
+};
+
+/**
+ * Query the state of a Hashed Time Lock Contract using contractId
+ * - Determine if a unique asset instance is locked by a given party for another given party
+ **/
+const isAssetLockedInHTLCqueryUsingContractId = async (
+    contract: Contract,
+    contractId: string,
+): Promise<any> => {
```

Comment:
fix return type

Reply:
@outsidethecode Please replace with unknown as per the earlier suggestion.

Reply:
I was following the exact pattern in `AssetManager.ts`. Anyway, I've changed it to `boolean`.